scripts/create_initial_tags.py

```diff
@@ -0,0 +1,250 @@
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from flask_appbuilder import Model
+from sqlalchemy import create_engine
+from sqlalchemy.sql import func, functions, join, literal, select, union_all
+
+from superset import app
+from superset.models.tags import ObjectTypes, TagTypes
+
+
+def add_types(engine, metadata):
+    """
+    Tag every object according to its type:
+
+      INSERT INTO tagged_object (tag_id, object_id, object_type)
+      SELECT
+        tag.id AS tag_id,
+        slices.id AS object_id,
+        'chart' AS object_type
+      FROM slices
+      JOIN tag
+      ON tag.name = 'type:chart'
+      UNION ALL
+      SELECT
+        tag.id AS tag_id,
+        dashboards.id AS object_id,
+        'dashboard' AS object_type
+      FROM dashboards
+      JOIN tag
+      ON tag.name = 'type:dashboard'
+      UNION ALL
+      SELECT
+        tag.id AS tag_id,
+        saved_query.id AS object_id,
+        'query' AS object_type
+      FROM saved_query
+      JOIN tag
+      ON tag.name = 'type:query';
+
+    """
+
+    tag = metadata.tables['tag']
+    tagged_object = metadata.tables['tagged_object']
+    slices = metadata.tables['slices']
+    dashboards = metadata.tables['dashboards']
+    saved_query = metadata.tables['saved_query']
+
+    # add a tag for each object type
+    insert = tag.insert()
+    for type in ObjectTypes.__members__:
+        engine.execute(insert, name=f'type:{type}', type=TagTypes.type)
+
+    charts = select([
+        tag.c.id.label('tag_id'),
+        slices.c.id.label('object_id'),
+        literal(ObjectTypes.chart.name).label('object_type'),
+    ]).select_from(join(slices, tag, tag.c.name == 'type:chart'))
+
+    dashboards = select([
+        tag.c.id.label('tag_id'),
+        dashboards.c.id.label('object_id'),
+        literal(ObjectTypes.dashboard.name).label('object_type'),
+    ]).select_from(join(dashboards, tag, tag.c.name == 'type:dashboard'))
+
+    saved_queries = select([
+        tag.c.id.label('tag_id'),
+        saved_query.c.id.label('object_id'),
+        literal(ObjectTypes.query.name).label('object_type'),
+    ]).select_from(join(saved_query, tag, tag.c.name == 'type:query'))
+
+    combined = union_all(charts, dashboards, saved_queries).alias().select()
```

Comment:
Wouldn't 3 transactions instead of a bigger one be better? Not a big difference, but would allow to break down in 3 smaller chunks. If one section takes a long time or fails, we'd have more clarity as to where it's coming from.

Reply:
You're right, especially if the script can be run multiple times. Originally I ran this with Superset stopped, so it wasn't a problem. I'll fix it.